application/src/components/views/CityView.tsx

```diff
@@ -19,27 +21,46 @@ const CityView: React.FC<CityViewProps> = () => {
     setCurrentCityInfo({
       cityHall: { name: 'City Hall', level: 10, built: true },
       barracks: { name: 'Barracks', level: 2, built: true },
-      forge: { name: 'Forge', level: 0, built: false },
+      forge: { name: 'Forge', level: 0, built: false},
     });
   }, []);
 
-  const _renderBuildingIfBuilt = (currentBuildingInfo: buildingInfo) => {
+  const _renderBuildingIfBuilt = (currentBuildingInfo: buildingInfo, buildingId:string) => {
```

Comment:
The rendering of the whole building should be here, you can also use a new visual component to represent buildings